### ISSUE TITLE
Bug: no-unsupported-features/es-builtins: fetch reported unsupported despite Node.js >=18.16.0

### ISSUE THREAD

**Issue by caugner** [eslint-community/eslint-plugin-n#253]
### Environment

Node version: 18.20.1
npm version: 10.5.0
ESLint version: 8.57.0
eslint-plugin-n version: 17.3.0
Operating System: Mac OS / Ubuntu 22.04

### What rule do you want to report?

n/no-unsupported-features/es-builtins

### Link to Minimal Reproducible Example

(unclear how to reproduce this on the ESLint playground)

### What did you expect to happen?

I expected the following to not cause any errors:

```sh
git clone -b dependabot/npm_and_yarn/eslint-plugin-n-17.3.0 git@github.com:mdn/yari.git --depth=1
cd yari
yarn install
yarn eslint .
```

But actually I'm seeing these errors, although our `package.json` requires Node.js  >=18.16.0:

```
  30:31  error  The 'fetch' is still an experimental feature and is not supported until Node.js 18.0.0. The configured version range is '>=18.0.0'  n/no-unsupported-features/es-builtins
  36:34  error  The 'fetch' is still an experimental feature and is not supported until Node.js 18.0.0. The configured version range is '>=18.0.0'  n/no-unsupported-features/es-builtins
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Noticed via failing test in [this Dependabot PR](https://github.com/mdn/yari/pull/10993).

**Comment by scagood** [comment]
Thank you for the report!

:thinking: Technically fetch was not `supported` until `21.0.0`, and so should be `experimental` in `18.20.1`:
[![image](https://github.com/eslint-community/eslint-plugin-n/assets/2230835/a60f62ff-618e-4ea1-92b2-40ef74e2ca78)](https://nodejs.org/docs/latest-v21.x/api/globals.html#fetch)

---

However, there is still a bug here, as we mark is as supported in v18:
https://github.com/eslint-community/eslint-plugin-n/blob/a0b45ee9c35279c29d3602609dad3b36d95d6f5f/lib/rules/no-unsupported-features/es-builtins.js#L604-L606

This could be related to #252, but `18.0.0` is a subset of `>=18.0.0`.

**Comment by scagood** [comment]
It makes me think that the wrong value is getting to [`isSupported`](https://github.com/eslint-community/eslint-plugin-n/blob/master/lib/util/check-unsupported-builtins.js#L35C37-L35C47)

As the range is truthy:
https://runkit.com/scagood/662900db83c1e60008f2ccd4

[![image](https://github.com/eslint-community/eslint-plugin-n/assets/2230835/e0b6f9af-84de-4826-aef7-a6507714f71d)](https://runkit.com/scagood/662900db83c1e60008f2ccd4)

**Comment by aladdin-add** [comment]
```bash
> semverRangeSubset('>=18.1.0', '>=18.0.0 <19||>18')
false
```

it seems an issue in `node-semver`. I'll revert that change then.

**Comment by aladdin-add** [comment]
just pushed a patch: https://github.com/eslint-community/eslint-plugin-n/releases/tag/v17.3.1

**Comment by scagood** [comment]
mmm, that is a strange bug!


```js
const subset = require("semver/ranges/subset")

subset('^17.2.0 || >17', '^17.2.0 || >17')  // true
subset('^17.2.0 || >17', '^17.2.0 || >=18') // true
subset('^17.2.0 || >=18', '^17.2.0 || >17') // true

subset('^17.2.0 || >17', '>=17.2.0') // true
subset('>=17.2.0', '^17.2.0 || >17') // false

subset('^17.2.0 || >=18', '>=17.2.0') // true
subset('>=17.2.0', '^17.2.0 || >=18') // false
```

**Comment by aladdin-add** [comment]
can you file the issue in `node-semver` repo, please? 🙏

**Comment by scagood** [comment]
I am going to close this, as I think the patch did fix it.

We can track [the semver issue](https://github.com/npm/node-semver/issues/703) in #250.
